### PR TITLE
Multiselect UI logic

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/models/thread/SelectedThread.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class SelectedThread(thread: Thread) {
+    val uid = thread.uid
+    val isFavorite = thread.isFavorite
+    val unseenMessagesCount = thread.unseenMessagesCount
+
+    override fun equals(other: Any?): Boolean = other is SelectedThread && uid == other.uid
+
+    override fun hashCode(): Int {
+        return uid.hashCode()
+    }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/models/thread/SelectedThread.kt]:
```suggestion
    override fun hashCode(): Int = uid.hashCode()
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 2]
```diff
@@ -63,8 +66,10 @@ class ThreadListAdapter(
     private var folderRole: FolderRole?,
     private var contacts: Map<Recipient, MergedContact>,
     private val onSwipeFinished: () -> Unit,
+    private val multiSelection: MultiSelectionListener<SelectedThread>? = null,
 ) : DragDropSwipeAdapter<Any, ThreadViewHolder>(mutableListOf()), RealmChangesBinding.OnRealmChanged<Thread> {
 
+    private var threadCount = -1
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:72]: Confusing name with `threadCount` views.

File: app/src/main/res/menu/multiselection_menu.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~ Infomaniak kMail - Android
+  ~ Copyright (C) 2022 Infomaniak Network SA
```
Comment (KevinBoulongne) [inline app/src/main/res/menu/multiselection_menu.xml:3]:
```suggestion
  ~ Copyright (C) 2023 Infomaniak Network SA
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt [hunk 4]
```diff
@@ -160,6 +165,13 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
             folderRole = mainViewModel.currentFolder.value?.role,
             contacts = mainViewModel.mergedContacts.value ?: emptyMap(),
             onSwipeFinished = { isRecoveringFinished.value = true },
+            multiSelection = object : MultiSelectionListener<SelectedThread> {
+                override var isEnabled by mainViewModel::isMultiSelectOn
+                override val selectedItems by mainViewModel::selectedThreads
+                override fun publishSelectedItems() {
+                    mainViewModel.selectedThreadsLiveData.value = mainViewModel.selectedThreadsLiveData.value
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt:172]:
```suggestion
                    mainViewModel.selectedThreadsLiveData.value = mainViewModel.selectedThreads
```

Comment (sirambd) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt:172]: move to the viewModel

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**